bindings/js/api.js

```diff
@@ -144,114 +276,346 @@ var HaclWasm = (function() {
         instead 'name' will contain the full name of the function
   */
 
-  var check_if_byte_array = function(candidate, length, name) {
+  var array_type = function(type) {
+    switch (type) {
+      case "buffer":
+        return Uint8Array;
+      case "buffer(uint32)":
+        return Uint32Array;
+      case "buffer(uint64)":
+        return BigUint64Array;
+      default:
+        throw new Error("Unknown array type: "+type);
+    }
+  }
+
+  var cell_size = function(type) {
+    switch (type) {
+      case "buffer":
+        return 1;
+      case "buffer(uint32)":
+        return 4;
+      case "buffer(uint64)":
+        return 8;
+      default:
+        throw new Error("Unknown array type: "+type);
+    }
+  }
+
+  var check_array_type = function(type, candidate, length, name) {
     if ((typeof(candidate) !== "object") ||
       (candidate.length !== length) ||
-      (candidate.constructor !== Uint8Array)
+      (candidate.constructor !== array_type(type))
```

Comment:
`typeof candidate !== "object" || candidate.constructor !== array_type(type)` could be simplified into `!(candiate instanceof array_type(type))` here